app/services/preserved_object_handler.rb

```diff
@@ -60,7 +60,19 @@ def create
   end
 
   def check_existence
-    # this is just a stub
+    if invalid?
+      handler_results.add_result(PreservedObjectHandlerResults::INVALID_ARGUMENTS, errors.full_messages)
+    elsif PreservedObject.exists?(druid: druid)
+      confirm_version
```

Comment:
If we call confirm_version immediately, it will do the argument validation, and it avoids the extra call to "PreservedObject.exists?" as well. Another approach would be to call confirm_version itself but to call the private methods it uses.